relay-cardinality/src/limiter.rs

```diff
@@ -20,19 +19,27 @@ pub struct Scoping {
     pub project_id: ProjectId,
 }
 
+/// Accumulator of all cardinality outcomes.
+pub trait Outcomes {
+    /// Called for ever [`Entry`] which was rejected from the [`Limiter`].
+    fn reject(&mut self, entry_id: EntryId);
+}
+
 /// Limiter responsible to enforce limits.
 pub trait Limiter {
     /// Verifies cardinality limits.
     ///
     /// Returns an iterator containing only accepted entries.
-    fn check_cardinality_limits<I>(
+    fn check_cardinality_limits<I, T>(
         &self,
         scoping: Scoping,
         limits: &[CardinalityLimit],
         entries: I,
-    ) -> Result<Box<dyn Iterator<Item = Rejection>>>
+        outcomes: &mut T,
```

Comment:
I initially had this as `&mut dyn Outcomes`, get's rid of the generic at some dynamic dispatch. Not sure what to prefer here.